src/login/logind-seat.c

```diff
@@ -376,17 +375,9 @@ int seat_read_active_vt(Seat *s) {
         t[k] = 0;
         truncate_nl(t);
 
-        if (!startswith(t, "tty")) {
-                log_error("Hm, /sys/class/tty/tty0/active is badly formatted.");
-                return -EIO;
-        }
-
-        r = safe_atou(t+3, &vtnr);
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse VT number \"%s\": %m", t+3);
-
-        if (!vtnr) {
-                log_error("VT number invalid: %s", t+3);
+        vtnr = vtnr_from_tty(t);
+        if (vtnr < 0) {
+                log_error_errno(vtnr, "Hm, /sys/class/tty/tty0/active is badly formatted: %m");
                 return -EIO;
```

Comment:
It seems the returned value `-EIO` is never checked. So, `return log_error_errno(...)` is simpler, I think.

Reply:
I thought about this too but preferred to leave this "unrelated" change to a separate commit if it's needed. This commit is just for making use of vtnr_from_tty() to make the code shorter with no functional changes.